src/core/cuda_wrapper.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
```

Comment:
prefer include guards since we are using it all over and it's not part of the standard?